service/src/main.rs

```diff
@@ -677,38 +673,32 @@ fn execute_trusted_calls<E: Sidechain>(enclave_api: &E) {
 }
 
 /// gets a list of blocks that need to be synced, ordered from oldest to most recent header
-/// blocks that need to be synced are all blocks from the current header to the last synced header, iterating over parent
-fn get_blocks_to_sync(
+/// starting from the last synced head until the current head but at most BLOCK_SYNC_BATCH_SIZE.
+fn get_block_chunk_to_sync(
 	api: &Api<sr25519::Pair, WsRpcClient>,
 	last_synced_head: &Header,
 	curr_head: &SignedBlock,
 ) -> Vec<SignedBlock> {
 	let mut blocks_to_sync = Vec::<SignedBlock>::new();
 
-	// add blocks to sync if not already up to date
-	if curr_head.block.header.hash() != last_synced_head.hash() {
-		blocks_to_sync.push((*curr_head).clone());
-
-		// Todo: Check, is this dangerous such that it could be an eternal or too big loop?
-		let mut head = (*curr_head).clone();
-		let no_blocks_to_sync = head.block.header.number - last_synced_head.number;
-		if no_blocks_to_sync > 1 {
-			println!("light client is synced until block: {:?}", last_synced_head.number);
-			println!("Last finalized block number: {:?}\n", head.block.header.number);
-		}
-		while head.block.header.parent_hash != last_synced_head.hash() {
-			debug!("Getting head of hash: {:?}", head.block.header.parent_hash);
-			head = api.signed_block(Some(head.block.header.parent_hash)).unwrap().unwrap();
-			blocks_to_sync.push(head.clone());
-
-			if head.block.header.number % BLOCK_SYNC_BATCH_SIZE == 0 {
-				println!(
-					"Remaining blocks to fetch until last synced header: {:?}",
-					head.block.header.number - last_synced_head.number
-				)
-			}
+	let mut head = curr_head.clone();
+	let no_blocks_to_sync = curr_head.block.header.number - last_synced_head.number;
+	if no_blocks_to_sync > 1 {
+		println!("light client is synced until block: {:?}", last_synced_head.number);
+		println!("Last finalized block number: {:?}\n", head.block.header.number);
+	}
+	for n in (last_synced_head.number + 1)..curr_head.block.header.number {
+		debug!("Getting head of hash: {:?}", head.block.header.parent_hash);
+		head = api.get_signed_block_by_num(Some(n)).unwrap().unwrap();
+		blocks_to_sync.push(head.clone());
+
+		if head.block.header.number != 0 && head.block.header.number % BLOCK_SYNC_BATCH_SIZE == 0 {
+			println!(
+				"Remaining blocks to fetch until last synced header: {:?}",
+				head.block.header.number - last_synced_head.number
+			);
+			return blocks_to_sync
```

Comment:
As advised above, I think we should make this function agnostic to the batch size.